### PR TITLE
find_partition: provide some partitioning stats

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: pytato/partition.py [hunk 1]
```diff
@@ -340,6 +342,16 @@ def find_partition(outputs: DictOfNamedArrays,
     if __debug__:
         _check_partition_disjointness(result)
 
+    from pytato.analysis import get_num_nodes
+    num_nodes = []
```
Comment (inducer) [inline pytato/partition.py:346]:
```suggestion
    num_nodes_per_part = []
```

Comment (inducer) [inline pytato/partition.py:346]: Make this a list comprehension.

File: pytato/partition.py [hunk 2]
```diff
@@ -340,6 +342,16 @@ def find_partition(outputs: DictOfNamedArrays,
     if __debug__:
         _check_partition_disjointness(result)
 
+    from pytato.analysis import get_num_nodes
```
Comment (inducer) [inline pytato/partition.py:345]: Add a folding comment around this to make the code more skimmable.

**Review by inducer — APPROVED** [review]
LGTM, thanks! Just a few style things to fix.

File: pytato/partition.py (hunk 2 above)
Comment (matthiasdiener) [reply pytato/partition.py:347]: Done in [e375c9d](https://github.com/inducer/pytato/pull/270/commits/e375c9d7f04d8682b08512217b2efff17bb7e0bc)

File: pytato/partition.py (hunk 1 above)
Comment (matthiasdiener) [reply pytato/partition.py]: Done in [e375c9d](https://github.com/inducer/pytato/pull/270/commits/e375c9d7f04d8682b08512217b2efff17bb7e0bc)

Comment (matthiasdiener) [reply pytato/partition.py]: Done in [e375c9d](https://github.com/inducer/pytato/pull/270/commits/e375c9d7f04d8682b08512217b2efff17bb7e0bc)

**Pull request merged**